minhton/src/algorithms/leave/minhton_leave_algorithm.cpp

```diff
@@ -58,8 +58,11 @@ void MinhtonLeaveAlgorithm::processFindReplacement(const minhton::MessageFindRep
     event_id = message.getHeader().getEventId();
   }
   this->access_->procedure_info->saveEventId(ProcedureKey::kLeaveProcedure, event_id);
+  find_end_helper_.setHopCount(message.getHopCount());
+
   this->performFindReplacement(message);
   this->access_->procedure_info->removeEventId(ProcedureKey::kLeaveProcedure);
+  find_end_helper_.resetHopCount();
```

Comment:
Even if this might require more code: Could this become more stateless by directly passing the hop count as a parameter to methods requiring it? Setting the hop count as a member variable before calling the algorithm and clearing it afterwards might be error-prone.